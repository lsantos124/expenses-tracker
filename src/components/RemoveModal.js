import React from 'react';
import Modal from 'react-modal';

const RemoveModal = (props) => (
	<Modal
		isOpen={props.delete}
		contentLabel="Are you sure you want to remove this expense?"
		onRequestClose={props.handleCancelDeleteExpense}
		closeTimeoutMS={200}
		className="modal"
		ariaHideApp={false}
	>
		<h3 className="modal__title">Are you sure you want to remove this expense?</h3>
		<button className="button button--modal" onClick={props.handleDeleteExpense}>Yes</button>
		<button className="button button--modal" onClick={props.handleCancelDeleteExpense}>No</button>
	</Modal>
);

export default RemoveModal;