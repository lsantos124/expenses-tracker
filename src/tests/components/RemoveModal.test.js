import React from 'react';
import { shallow } from 'enzyme';
import RemoveModal from '../../components/RemoveModal';

let handleDeleteExpense, handleCancelDeleteExpense, wrapper;

beforeEach(() => {
	handleDeleteExpense = jest.fn();
	handleCancelDeleteExpense = jest.fn();
	wrapper = shallow(
		<RemoveModal 
			delete={true}
			handleDeleteExpense={handleDeleteExpense} 
			handleCancelDeleteExpense={handleCancelDeleteExpense} 
		/>
	);
});

test('should render React Modal component correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

test('should call handleDeleteExpense', () => {
	wrapper.find('button').at(0).simulate('click');
	expect(handleDeleteExpense).toHaveBeenCalled();
});

test('should call handleCancelDeleteExpense', () => {
	wrapper.find('button').at(1).simulate('click');
	expect(handleCancelDeleteExpense).toHaveBeenCalled();
});