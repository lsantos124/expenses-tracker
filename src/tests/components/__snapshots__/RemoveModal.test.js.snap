// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render React Modal component correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Are you sure you want to remove this expense?"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Are you sure you want to remove this expense?
  </h3>
  <button
    className="button button--modal"
    onClick={[Function]}
  >
    Yes
  </button>
  <button
    className="button button--modal"
    onClick={[Function]}
  >
    No
  </button>
</Modal>
`;
